lib/ansible/plugins/callback/telegraf.py

```diff
@@ -0,0 +1,290 @@
+      - grep --binary-file=text changed /var/log/telegraf.log to see if telegraf is receiving 
+      - ensure telegraf is running with correct input blocks in /etc/telegraf/telegraf.d/
+        inputs-socket-listener.conf or inputs-http-listener.conf https://github.com/influxdata/telegraf/tree/master/plugins/inputs/socket_listener
+    options:
+      host:
+        required: True
+        description: telegraf host reachable from the ansible controller
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_HOST
+      metric_prefix:
+        description: metrics prefix for all metrics sent to telegraf.
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_PREFIX
+        default: "ansible."
+        ini:
+          - section: callback_telegraf
+            key: metric_prefix
+      metric_per_host:
+        description: summarize over all tasks per host
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_PER_HOST
+        default: True
+        ini:
+          - section: callback_telegraf
+            key: metric_per_host
+      metric_summary:
+        description: summarize stats over all hosts.
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_METRIC_SUMMARY
+        default: False
+        ini:
+          - section: callback_telegraf
+            key: metric_summary
+      wire_protocol:
+        description: http, https or udp.  If http you must pip install pytelegraf[http]. Default udp
+        choices: ['http', 'https', 'udp']
+        env:
+          - name: ANSIBLE_TELEGRAF_CALLBACK_WIRE_PROTOCOL
+        default: http
+        ini:
+          - section: callback_telegraf
+            key: wire_protocol
+'''
+
+import getpass
+import os.path
+import socket
+from datetime import datetime, timedelta
+import yaml
+
+
+try:
+    # Ansible v2
+    from ansible.plugins.callback import CallbackBase
+    from __main__ import cli
+except ImportError:
+    # Ansible v1
+    CallbackBase = object
+    cli = None
+
+
+
+
+class CallbackModule(CallbackBase):
+    """
+    This is an ansible callback plugin that sends metrics
+    to a collector via telegraf.  Which collector depends on telegraf config.
+    """    
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'metrics'  # https://groups.google.com/forum/#!topic/ansible-devel/G9mJUmaM-Eg
+    CALLBACK_NAME = 'telegraf'
+    CALLBACK_NEEDS_WHITELIST = False
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+        self._display.display('Playbook telegraf_callback init')
+
+        # the ansible controller running the callback
+        self._controller_hostname = socket.gethostname()
+
+
+        # Read config and set up API client
+        conf_path = os.path.join(os.path.dirname(__file__), 'telegraf_callback.yml')
+        params = self._load_conf(conf_path)
+        
+        # override config values with ENV vars if set
+        host = params.get('host', 'localhost')
```

Comment:
this whole section is not needed, just use `self._plugin_options[<option name>]`

Reply:
self._plugin_options is none.  Trying to understand how it works from other callbacks.

Reply:
its new in 2.4 so it doesn't apply to older versions